go/tools/gopackagesdriver/bazel_json_builder.go

```diff
@@ -107,6 +97,20 @@ func (b *BazelJSONBuilder) localQuery(request string) string {
 	return fmt.Sprintf(`kind("^(%s) rule$", %s)`, b.getKind(), request)
 }
 
+func (b *BazelJSONBuilder) adjustToRelativePathIfPossible(request string) string {
+	// If request is a relative path and gopackagesdriver is ran within a subdirectory of the
+	// workspace, we must first resolve the absolute path for it.
+	// Note: Using FromSlash/ToSlash for handling windows
+	absRequest := filepath.FromSlash(request)
+	if !filepath.IsAbs(absRequest) {
+		absRequest = filepath.Join(b.bazel.BuildWorkingDirectory(), absRequest)
+	}
+	if relPath, err := filepath.Rel(workspaceRoot, absRequest); err == nil {
```

Comment:
What would be a possible error case here? Should we log it?

Reply:
I am looking at the code: https://pkg.go.dev/path/filepath#Rel

There seems to be two cases. One is on windows if you give paths on different roots (like c:\a\b z:\y\x, you can't make it relative because you don't share a root. The other case I found is if to resolve you would need th current directory: filepath.Rel("./b/c", "/a"). This would not happen if we could guarantee that both of them are Abs paths.

From what I can tell, bazel sets env BUILD_WORKSPACE_DIRECTORY and BUILD_WORKING_DIRECTORY is absolute already.

Neither case is a likely scenario here, so for now I will follow the suggestion of logging it.